astropy/units/function/logarithmic.py

```diff
@@ -277,6 +277,65 @@ def __isub__(self, other):
         self._set_unit(new_unit)
         return self
 
+    def __mul__(self, other):
+        # Multiply by a float or a dimensionless quantity
+        try:
+            other = float(other)
```

Comment:
Hmm, I think the issue with the failing test is that this means one can do `u.Magnitude(1*u.Jy) * u.Magnitude(2)`, which would be interpreted as `u.Magnitude(1*u.Jy) * 0.15848931924611134` (because `float(u.Magnitude(2))` works).

That suggests perhaps writing the `try/except` differently, as
```
try:
    new_physical_unit = self.unit.physical_unit**other
except ...:
    ...
```
In a way, that is perhaps cleaner anyway?